src/rgw/rgw_auth_filters.h

```diff
@@ -238,6 +238,7 @@ class SysReqApplier : public DecoratedApplier<T> {
   const RGWHTTPArgs& args;
   mutable boost::tribool is_system;
   mutable std::optional<ACLOwner> effective_owner;
+  mutable std::string effective_tenant;
```

Comment:
this should probably be `std::optional<std::string>` so we can distinguish between an empty `effective_tenant` (which `get_tenant()` should return directly) and the lack of an `rgwx-uid` where we should fall back to `DecoratedApplier<T>::get_tenant()`